test-suite/src/test/scala/org/scalajs/testsuite/junit/ScalaJSJUnitAssertionsTest.scala

```diff
@@ -0,0 +1,304 @@
+        equals: Boolean = true): Unit = {
+      testIfAsserts(assertArrayEquals(message, expected, actual), equals)
+      testIfAsserts(assertArrayEquals(expected, actual), equals)
+    }
+    def testIntAssertion(expected: Array[Int], actual: Array[Int],
+        equals: Boolean = true): Unit = {
+      testIfAsserts(assertArrayEquals(message, expected, actual), equals)
+      testIfAsserts(assertArrayEquals(expected, actual), equals)
+    }
+    def testLongAssertion(expected: Array[Long], actual: Array[Long],
+        equals: Boolean = true): Unit = {
+      testIfAsserts(assertArrayEquals(message, expected, actual), equals)
+      testIfAsserts(assertArrayEquals(expected, actual), equals)
+    }
+    def testDoubleAssertion(expected: Array[Double], actual: Array[Double],
+        delta: Double, equals: Boolean = true): Unit = {
+      testIfAsserts(assertArrayEquals(message, expected, actual, delta), equals)
+      testIfAsserts(assertArrayEquals(expected, actual, delta), equals)
+    }
+    def testFloatAssertion(expected: Array[Float], actual: Array[Float],
+        delta: Float, equals: Boolean = true): Unit = {
+      testIfAsserts(assertArrayEquals(message, expected, actual, delta), equals)
+      testIfAsserts(assertArrayEquals(expected, actual, delta), equals)
+    }
+
+    // Array tests
+    testAnyRefAssertion(arr1, arr1)
+    testAnyRefAssertion(Array(obj1), Array(obj1))
+    testAnyRefAssertion(Array(obj1, obj2, obj2), Array(obj1, obj2, obj2))
+    testAnyRefAssertion(Array(obj1), Array("0"))
+    testAnyRefAssertion(Array(Array(1), Array(2, Array(3))),
+      Array(Array(1), Array(2, Array(3))))
+    testIntAssertion(Array(1, 2, 3), Array(1, 2, 3))
+    testLongAssertion(Array(1L, 2L, 3L), Array(1L, 2L, 3L))
+
+    testAnyRefAssertion(Array(obj1), Array(obj2), notEquals)
+    testAnyRefAssertion(Array(obj1, obj2, obj2), Array(obj1, obj2, obj1), notEquals)
+    testAnyRefAssertion(Array(obj1), Array("4"), notEquals)
+    testAnyRefAssertion(Array(Array(2), Array(2, Array(3))), Array(Array(1),
+      Array(2, Array(3))), notEquals)
+    testAnyRefAssertion(Array(Array(1, 2), Array(2, Array(3))),
+      Array(Array(1), Array(2, Array(3))), notEquals)
+    testAnyRefAssertion(Array(Array(1), Array(2, Array(3))),
+      Array(Array(1, 4), Array(2, Array(3))), notEquals)
+    testIntAssertion(Array(1, 2, 3), Array(1, 3, 3), notEquals)
+    testLongAssertion(Array(1L, 2L, 3L), Array(1L, 1L, 3L), notEquals)
+
+    // Array[Double]
+    testDoubleAssertion(Array(1d, 2d, 3d), Array(1d, 2d, 4d), 1d)
+    testDoubleAssertion(Array(1d, 2d, 3d), Array(1d, 2d, 3.5d), 1d)
+
+    // Array[Float]
+    testFloatAssertion(Array(1f, 2f, 3f), Array(1f, 2f, 4f), 1f)
+    testFloatAssertion(Array(1f, 2f, 3f), Array(1f, 2f, 3.5f), 1f)
+  }
+
+  @Test
+  def testAssertThat(): Unit = {
+    testIfAsserts(assertThat("42", instanceOf[String](classOf[String])))
+    testIfAsserts(assertThat("42", instanceOf[String](classOf[Int])), shallNotPass)
```

Comment:
The problem here is that the definition of `isInstanceOf` is `def instanceOf[T](typ: Class[_]): Matcher[T]` and therefore one could potentially write `instanceOf[String](classOf[Int]))`, but it should fail the assertion. I added the types to make it explicit on these corner cases.